doc/Hammer-Use/Hierarchical.rst

```diff
@@ -1,2 +1,106 @@
+
+Cadence Implementation
+----------------------
+
+Currently, the hierarchical flow is implemented with the Cadence plugin using its Interface Logic Model (ILM) methodology. At the end of each submodule's place-and-route, an ILM is written as the hardened macro, which contains an abstracted view of its design and timing models only up to the first sequential element.
+
+ILMs are similar to LEFs and LIBs for traditional hard macros, except that the interface logic is included in all views. This means that at higher levels of hierarchy, the ILM instances can be flattened at certain steps, such as those that perform timing analysis on the entire design, resulting in a more accurate picture of timing than a normal LIB would afford.
+
+Tips for Constraining Hierarchical Modules
+------------------------------------------
+
+In a bottom-up hierarchical flow, is is important to remember that submodules do not know the environment in which they will placed. This means:
+
+* At minimum, the pins must be placed on the correct edges of the submodule on metal layers that are accessible in the parent level. Depending on the technology, this may interfere with things like power straps near the edge, so custom obstructions may be necessary. If fixed IOs are placed in submodules (e.g. bumps), then in the parent level, those pins must be promoted up using the ``preplaced: true`` option in the pin assignment.
+
+* Clocks should be constrained more tightly for a submodule compared to its parent to account for extra clock insertion delay, jitter, and skew at increasingly higher levels of hierarchy. Otherwise, you may run into surprise timing violations in submodule instances even if those passed timing in isolation.
+
+* You may need to specify pin delays ``vlsi.inputs.delays`` for many pins to optimize the partitioning of sequential signals that cross the submodule boundary. By default, without pin delay constraints, the input and output delay are constrained to be coincident with its related clock arrival at the module boundary.
+
+* Custom SDC constraints that originate from a higher level (e.g. false paths from async inputs) need to be specified in submodules as well.
+
+* Custom CPFs will need to be written if differently-named power nets need to globally connected between submodules. Similarly, hierarchical flow with custom CPFs can also be used to fake a multi-power domain topology until Hammer properly supports this feature.
+
+Special Notes & Limitations
+---------------------------
+
+#. Hammer IR keys propagate up through the hierarchical tree. For example, if ``vlsi.inputs.clocks`` was specified in the constraints for ``ModuleAA`` but not for ``ModuleA``, ``ModuleA`` will inherit ``ModuleAA``'s constraints. Take special care of where your constraints come from, especially for a parent module with more than one submodule.
```

Comment:
Are there cases where this behavior is useful? 

It would also be helpful to include YAML snippets demonstrating the issue, for this and the next bullet point.

Reply:
I don't believe this behavior is actually useful, and these bullets are meant to warn against situations where observed results don't match against the input config. Fixing this aligns with our internal discussions to parse + propagate IR on a module-by-module basis in hierarchical mode. As such, I don't see the value in including a YAML example snippet for these, but I will update this to recommend explicitly writing all the same keys for every module to avoid any confusion.